activation/atxlistener.go

```diff
@@ -0,0 +1,71 @@
+package activation
+
+import (
+	"context"
+	"sync"
+
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/p2p/service"
+	"github.com/spacemeshos/go-spacemesh/priorityq"
+	"gopkg.in/tomb.v2"
+)
+
+// AtxProtocol is the protocol id for broadcasting atxs over gossip
+const AtxProtocol = "AtxGossip"
+
+type atxProcessor interface {
+	HandleGossipAtx(context.Context, service.GossipMessage, service.Fetcher)
+}
+
+// ATXListener handles ATX gossip messages.
+type ATXListener struct {
+	startOnce   sync.Once
+	tomb        tomb.Tomb
```

Comment:
are you sure that tomb makes sense here? it looks like it is the same as errgroup and supposed to manage goroutines that must be killed if one of them exits with error

from doc

> If any of the tracked goroutines returns a non-nil error, or the Kill or Killf method is called by any goroutine in the system (tracked or not), the tomb Err is set, Alive is set to false, and the Dying channel is closed to flag that all tracked goroutines are supposed to willingly terminate as soon as possible.

Reply:
i see now that you always return nil